### PR TITLE
add replication delay command

### PR DESCRIPTION
**Pull request by marcosvm**
Related issue: https://github.com/openark/orchestrator/issues/1266

### Description

This PR adds a command to delay replication on a specific instance keeping the current replication threads status.

Please make sure that:

- [ ] contributed code is using same conventions as original code

Please make sure the PR passes CI tests. For your information, CI tests the following:

- [x] - code is formatted via `gofmt` (please avoid `goimports`)
- [x] - code passes compilation
- [x] - code passes unit tests
- [x] - code passes integration tests with MySQL backend
- [x] - code passes integration tests with SQLite backend
- [x] - There are no orphaned docs/ pages (there's some link in the docs to point to any page)
- [x] - upgrade from previous version (`master` branch) is successful

### CODE REVIEW

File: go/inst/instance_topology_dao.go [hunk 1]
```diff
@@ -595,6 +595,24 @@ func EnableSemiSync(instanceKey *InstanceKey, master, replica bool) error {
 	return err
 }
 
+// DelayReplication set the replication delay given seconds
+// keeping the current state of the replication threads.
+func DelayReplication(instanceKey *InstanceKey, seconds int) error {
```
Comment (shlomi-noach) [inline go/inst/instance_topology_dao.go:600]: Let's have this function validate `seconds` value is `>= 0`. I see it's been validated in the API, but it's important that the function self-validates its own input. If you like, you can then remove the ` seconds < 0 ` validation from https://github.com/openark/orchestrator/pull/1267/files#diff-535924bad43b8347b4337b267fbbef99095e909ac11bf5d458462a28d1afbf52R1592

File: go/http/api.go [hunk 2]
```diff
@@ -1577,6 +1577,31 @@ func (this *HttpAPI) DisableSemiSyncReplica(params martini.Params, r render.Rend
...
+		Respond(r, &APIResponse{Code: ERROR, Message: "Invalid value provided for seconds"})
+		return
+	}
+	err = inst.DelayReplication(&instanceKey, seconds)
+	if err != nil {
+		Respond(r, &APIResponse{Code: ERROR, Message: err.Error()})
+		return
+	}
+
+	Respond(r, &APIResponse{Code: OK, Message: fmt.Sprintf("Replication delayed: %+v", instanceKey), Details: nil})
```
Comment (shlomi-noach) [inline go/http/api.go]: May as well throw `seconds` into the output:
```suggestion
	Respond(r, &APIResponse{Code: OK, Message: fmt.Sprintf("Replication delayed: %+v", instanceKey), Details: seconds})
```

File: go/inst/instance_topology_dao.go [hunk 3]
```diff
@@ -595,6 +595,24 @@ func EnableSemiSync(instanceKey *InstanceKey, master, replica bool) error {
...
+	query := fmt.Sprintf("change master to master_delay=%d", seconds)
+	statements, err := GetReplicationRestartPreserveStatements(instanceKey, query)
+	if err != nil {
+		return err
+	}
+	for _, cmd := range statements {
+		if _, err := ExecInstance(instanceKey, cmd); err != nil {
+			return log.Errorf("%+v: DelayReplication: '%q' failed: %+v", *instanceKey, cmd, err)
+		} else {
+			log.Infof("%s on %+v as part of DelayReplication", cmd, *instanceKey)
```
Comment (shlomi-noach) [inline go/inst/instance_topology_dao.go]:
```suggestion
			log.Infof("DelayReplication: %s on %+v", cmd, *instanceKey)
```

File: go/inst/instance_topology_dao.go [hunk 4]
```diff
@@ -595,6 +595,24 @@ func EnableSemiSync(instanceKey *InstanceKey, master, replica bool) error {
...
+		}
+	}
+	return nil
```
Comment (shlomi-noach) [inline go/inst/instance_topology_dao.go:617]:
```suggestion
	AuditOperation("delay-replication", instanceKey, fmt.Sprintf("set to %d", seconds))
	return nil
```

**Review by shlomi-noach — CHANGES_REQUESTED** [review]
Thank you! This looks good, with a few requests for changes. Please consider adding a `delay-replication` command in https://github.com/openark/orchestrator/blob/master/resources/bin/orchestrator-client

**Comment by marcosvm** [comment]
@shlomi-noach I added the modifications you suggested. Thank you for the review.

**Review by shlomi-noach — APPROVED** [review]

**Pull request merged**